src/Interpreters/InterpreterShowIndexesQuery.cpp

```diff
@@ -0,0 +1,96 @@
+#include <Interpreters/InterpreterShowIndexesQuery.h>
+
+#include <IO/WriteBufferFromString.h>
+#include <Parsers/ASTShowIndexesQuery.h>
+#include <Parsers/formatAST.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/executeQuery.h>
+#include <Interpreters/parseDatabaseAndTableName.h>
+#include <IO/Operators.h>
+
+
+namespace DB
+{
+
+
+InterpreterShowIndexesQuery::InterpreterShowIndexesQuery(const ASTPtr & query_ptr_, ContextMutablePtr context_)
+    : WithMutableContext(context_)
+    , query_ptr(query_ptr_)
+{
+}
+
+
+String InterpreterShowIndexesQuery::getRewrittenQuery()
+{
+    const auto & query = query_ptr->as<ASTShowIndexesQuery &>();
+
+    auto [database, table] = parseDatabaseAndTableName(query, getContext()->getCurrentDatabase());
+
+    WriteBufferFromOwnString where_expression_buf;
+    if (query.where_expression)
+        where_expression_buf << "WHERE (" << query.where_expression << ")";
+    String where_expression = where_expression_buf.str();
+
+    WriteBufferFromOwnString rewritten_query;
+    rewritten_query     << "SELECT * FROM ("
+                        << "(SELECT "
+                            << "name AS table, "
+                            << "0 AS non_unique, "
+                            << "'PRIMARY' AS key_name, "
+                            << "NULL AS seq_in_index, "
+                            << "NULL AS column_name, "
+                            << "'A' AS collation, "
+                            << "NULL AS cardinality, "
+                            << "NULL AS sub_part, "
+                            << "NULL AS packed, "
+                            << "NULL AS null, "
+                            << "'primary' AS index_type, "
+                            << "NULL AS comment, "
+                            << "NULL AS index_comment, "
+                            << "'YES' AS visible, "
+                            << "primary_key AS expression "
+                        << "FROM system.tables "
+                        << "WHERE "
+                            << "database = '" << database << "' "
```

Comment:
SQL injection.

Reply:
I tried a few injection attacks from [here](https://www.w3schools.com/sql/sql_injection.asp) but either the parser rejected such queries or there was an error when the rewritten query was executed.

What worked was an attack with `1=1`, e.g. `show columns from tbl where field = 'b' or 1=1`. Since we would still be limited to system tables and not expose arbitrary table data, that should be okay as behavior.

Reply:
Just put into `backQuoteIfNeed`